test/test_images/probe_helper/probe_helper.go

```diff
@@ -352,5 +353,9 @@ func runProbeHelper(ctx context.Context, pubsubClient *pubsub.Client, storageCli
 	go sc.StartReceiver(ctx, forwardFromProbe(ctx, sc, psc, bkt, receivedEvents, timeout))
 	// Receive the event and return the result back to the probe
 	logger.Info("Starting event receiver...")
-	rc.StartReceiver(ctx, receiveEvent(ctx, receivedEvents))
+	go rc.StartReceiver(ctx, receiveEvent(ctx, receivedEvents))
+
+	if ready != nil {
```

Comment:
Is this needed?

Reply:
Not needed: I was trying to figure out a way to improve test flakiness, since there seemed to be a race between the probe helper starting and the testing client sending events to the probe helper, but this channel-based approach did not even really work after testing. Fixed this race by passing explicit listeners for the probe helper to use from the probe helper tests.